202308-quarto-websites/quarto-websites-notes.qmd

```diff
@@ -0,0 +1,190 @@
+-   Create website by publishing multiple documents into a unit
+    -   Specify layout, navigation, and style
+
+### RStudio overview
+
+-   Define RStudio
+    -   IDE = integrated development environment
+    -   Nice programs for writing code
+    -   RStudio is most common IDE for R
+    -   Others such as PyCharm or Visual Studio
+-   Layout
+    -   Upper left for displaying files that will make up website
+    -   Lower right hand "Files" tab shows folder structure and files on computer
+-   Projects
+    -   Organize work into self-contained units
+    -   Creates one folder with everything in it
+    -   Also solves working directory problems
+
+### Quarto documents
+
+-   These notes are an example
+-   Files with `.qmd` extension
+-   Next generation of R Markdown (`.rmd` files)
+-   What is Quarto?
+    -   Plain text files that are rendered to HTML
+    -   Specify formatting with special characters
+        -   Use markup language called markdown
+-   Why use Quarto?
+    -   Easy to read plain text files as-is
+    -   More control over formatting text files
+    -   Can switch back and forth between source and visual
+    -   Can intersperse with code (R or other languages)
+-   What are parts of a Quarto file?
+    -   Chunk at top
+        -   Contains file metadata
+        -   Uses another markup language called YAML
+        -   Will use in entire file later on
+    -   Body of file
+        -   Contains markdown
+-   Common markdown syntax
+    -   [Markdown Basics page](https://quarto.org/docs/authoring/markdown-basics.html)
+    -   Asterisks to bold or italicize text
+    -   Pound symbols for headers
+    -   Square brackets and parenthesis for links
+    -   Asterisks, dashes, or numbers for lists
+    -   Backticks for code
+-   View rendered document
+    -   Visual tab
+    -   Click "Render" button
+        -   Use gear button drop-down to choose where it shows up
+
+### Set up initial website
+
+-   [Quarto quickstart](https://quarto.org/docs/websites/#config-file)
+-   Can point-and-click to create skeleton of website
+-   Do with different platforms
+    -   RStudio is user-friendly point-and-click
+    -   Other IDEs
+    -   Command line, which is language-agnostic
+-   Create new project for website
+    -   "New Project" -\> "New Directory" -\> "Quarto Website"
+    -   Choose name and location for project files to be stored
+-   Recommend using visual markdown editor option
+    -   Renders files on the fly
+    -   Used to be buggy
+-   Creates files needed for skeleton of a website
+    -   `.Rproj` is for project
+    -   `_quarto.yml` is config file
+        -   Specify metadata about website and its different pages
+        -   Customize navigation, page layout, and theme
+    -   Two `.qmd` files
+        -   `about.qmd` & `index.qmd`
+        -   These are pages on the website
+-   See what website looks like
+    -   Open any `.qmd` file
+    -   Use "Render" button, just like with Quarto files
```

Comment:
There's also a"Render Webiste" button in the "Build" tab—not sure if that does something different but might be worth investigating

Reply:
Oh, awesome! I didn't know that was there. And it didn't seem great to have to open up a `.qmd` just to render it locally